tests/cbor_decode/test5_strange/src/main.c

```diff
@@ -192,37 +215,48 @@ void test_string_overflow(void)
 void test_optional(void)
 {
 	const uint8_t payload_optional1[] = {
-		0x83 /* List start */, 0xCA /* tag */, 0xF4 /* False */, 0x02, 0x03,
+		LIST(3) /* List start */, 0xCA /* tag */, 0xF4 /* False */, 0x02, 0x03,
+		END
 	};
 	const uint8_t payload_optional2[] = {
-		0x82 /* List start */, 0xCA /* tag */, 0xF4 /* False */, 0x03,
+		LIST(2) /* List start */, 0xCA /* tag */, 0xF4 /* False */, 0x03,
+		END
 	};
 	const uint8_t payload_optional3_inv[] = {
-		0x82 /* List start */, 0xCA /* tag */, 0xF4 /* False */, 0x02,
+		LIST(2) /* List start */, 0xCA /* tag */, 0xF4 /* False */, 0x02,
+		END
 	};
 	const uint8_t payload_optional4[] = {
-		0x83 /* List start */, 0xCA /* tag */, 0xF4 /* False */, 0x02, 0x01,
+		LIST(3) /* List start */, 0xCA /* tag */, 0xF4 /* False */, 0x02, 0x01,
+		END
 	};
 	const uint8_t payload_optional5[] = {
-		0x83 /* List start */, 0xCA /* tag */, 0xF5 /* True */, 0x02, 0x02,
+		LIST(3) /* List start */, 0xCA /* tag */, 0xF5 /* True */, 0x02, 0x02,
+		END
 	};
 	const uint8_t payload_optional6[] = {
-		0x84 /* List start */, 0xCA /* tag */, 0xF5 /* True */, 0xF4 /* False */, 0x02, 0x02,
+		LIST(4) /* List start */, 0xCA /* tag */, 0xF5 /* True */, 0xF4 /* False */, 0x02, 0x02,
+		END
 	};
 	const uint8_t payload_optional7_inv[] = {
-		0x82 /* List start */, 0xCB /* wrong tag */, 0xF4 /* False */, 0x03,
+		LIST(2) /* List start */, 0xCB /* wrong tag */, 0xF4 /* False */, 0x03,
+		END
 	};
 	const uint8_t payload_optional8_inv[] = {
-		0x82 /* List start */, 0xF4 /* False (no tag first) */, 0x03,
+		LIST(2) /* List start */, 0xF4 /* False (no tag first) */, 0x03,
+		END
 	};
 	const uint8_t payload_optional9[] = {
-		0x84 /* List start */, 0xCA /* tag */, 0xF4 /* False */, 0x02, 0x03, 0x08,
+		LIST(4) /* List start */, 0xCA /* tag */, 0xF4 /* False */, 0x02, 0x03, 0x08,
+		END
 	};
 	const uint8_t payload_optional10[] = {
-		0x86 /* List start */, 0xCA /* tag */, 0xF4 /* False */, 0x02, 0x03, 0x08, 0x08, 0x08,
+		LIST(6) /* List start */, 0xCA /* tag */, 0xF4 /* False */, 0x02, 0x03, 0x08, 0x08, 0x08,
+		END
 	};
 	const uint8_t payload_optional11_inv[] = {
-		0x86 /* List start */, 0xCA /* tag */, 0xF4 /* False */, 0x02, 0x03, 0x08, 0x08, 0x09,
+		LIST(6) /* List start */, 0xCA /* tag */, 0xF4 /* False */, 0x02, 0x03, 0x08, 0x08, 0x09,
```

Comment:
nit: I guess we can drop the `/* List start */` comments here now that we have the nice macro ;)

Reply:
Good point. removed.